internal/resolution/datasource/maven_registry.go

```diff
@@ -24,31 +24,95 @@ func NewMavenRegistryAPIClient(registry string) *MavenRegistryAPIClient {
 
 var errAPIFailed = errors.New("API query failed")
 
+// GetProject fetches a pom.xml specified by groupID, artifactID and version and parses it to maven.Project.
+// For a snapshot version, version level metadata is used to find the extact version value.
```

Comment:
Could you add some more information on how snapshot versions behave in maven (is there a doc you can link to?)

It took me a while (& looking at your tests & some maven repos) to try understand how it works.
From what I understand, when you depend on `pkg@1.2.3-SNAPSHOT`, it looks up in the registry `pkg/1.2.3-SNAPSHOT/maven-metadata.xml` to determine which file to actually download, and downloads it from `pkg/1.2.3-SNAPSHOT/1.2.3-realversion.pom`?

Reply:
Yes correct.

The closest official documentation I find is https://maven.apache.org/ref/3.9.9/maven-repository-metadata/ and  https://maven.apache.org/repositories/metadata.html but the link to V-level metadata is not found...

I added both links to the comment.